linkerd/app/outbound/src/http/logical/policy/route/backend/metrics.rs

```diff
@@ -26,14 +28,24 @@ pub fn layer<T, N>(
     metrics: &RouteBackendMetrics<T::StreamLabel>,
 ) -> impl svc::Layer<
     N,
-    Service = NewCountRequests<
-        ExtractRequestCount,
-        NewResponseDuration<T, ExtractRecordDurationParams<ResponseMetrics<T::StreamLabel>>, N>,
+    Service = NewRecordBodyData<
+        ExtractRecordBodyDataParams,
+        NewCountRequests<
+            ExtractRequestCount,
+            NewResponseDuration<T, ExtractRecordDurationParams<ResponseMetrics<T::StreamLabel>>, N>,
+        >,
```

Comment:
the additions to this function are the part of this change that spark the least joy.

this strikes me as one of the primary sharp edges we can consider sanding down in future refactoring work. my hunch is that we might be able to restructure this in a way where we can use `impl Service` syntax at function boundaries, and spare ourselves the need to name all of our successive layers (_and their accompanying parameter extractors and `NewService<T>` bounds_).

Reply:
weirdly, ci failed after i applied those changes here. not going to investigate that further rn, reverted.